dask/dataframe/shuffle.py

```diff
@@ -112,6 +116,40 @@ def __reduce__(self):
         ]
         return (SimpleShuffleLayer, tuple(getattr(self, attr) for attr in attrs))
 
+    def distributed_pack(self):
+        return {
+            "name": self.name,
+            "column": self.column,
+            "npartitions": self.npartitions,
+            "npartitions_input": self.npartitions_input,
+            "ignore_index": self.ignore_index,
+            "name_input": self.name_input,
+            "meta_input": self.meta_input.to_json(),
+            "parts_out": list(self.parts_out),
+        }
+
+    @classmethod
+    def distributed_unpack(cls, state, dsk, dependencies):
+        from distributed.utils import str_graph
+        from distributed.worker import dumps_task
+
+        state["meta_input"] = pd.read_json(state["meta_input"])
+        obj = cls(**state)
+        input_keys = set()
+        for v in obj._cull_dependencies(
+            [(obj.name, i) for i in range(obj.npartitions_input)]
+        ).values():
+            input_keys.update(v)
+
+        raw = dict(obj)
+        raw = str_graph(raw, extra_values=input_keys)
```

Comment:
I was a bit curious about the call to `str_graph` here.  I understand that we need to do this at some point but I was hoping that it could be general.  Thinking about this again though, this seems like the kind of thing that we do want to special-case in some circumstances (maybe we never make the tuples) so I guess it's maybe a good idea regardless.